torch/_dynamo/output_graph.py

```diff
@@ -2229,6 +2198,274 @@ def maybe_lift_tracked_freevar_to_input(self, arg):
+    #     For example, if an input to subgraph takes size [s1+s2//8], we'll look for the
+    #     the free symbols in the sizes and lift as inputs similar to 1 in _lift_symbols_in_symint)
+    # Do not call this function direclty. This is used in create_graph_input for
+    # automatically lifting the free basic symbols in example_value.
+    def _lift_basic_symbols(self, example_value, src: Optional[Source]):
+        # See note [Export inputs must be explicitly passed in]
+        is_strict_export = self.is_export
+        is_non_strict_export = torch.compiler.is_compiling()
+        if is_strict_export or is_non_strict_export:
+            return
+
+        if isinstance(example_value, torch.Tensor):
+            return self._lift_symbols_in_tensor(example_value, src)
+
+    # See Note [Auto lift basic free symbols when create_graph_input]
+    def _lift_symbols_in_tensor(self, t: torch.Tensor, src: Optional[Source]):
+        for i, s in enumerate(t.size()):
+            self._lift_symbols_in_symint(
+                s,
+                (
+                    TensorPropertySource(src, TensorProperty.SIZE, i)
+                    if src is not None
+                    else None
+                ),
+            )
+        if t.layout is torch.strided:
+            for i, s in enumerate(t.stride()):
+                self._lift_symbols_in_symint(
+                    s,
+                    (
+                        TensorPropertySource(src, TensorProperty.STRIDE, i)
+                        if src is not None
+                        else None
+                    ),
+                )
+            self._lift_symbols_in_symint(
+                t.storage_offset(),
+                (
+                    TensorPropertySource(src, TensorProperty.STORAGE_OFFSET)
+                    if src is not None
+                    else None
+                ),
+            )
+        elif t.layout is torch.sparse_coo:
+            self._lift_symbols_in_tensor(t._indices(), src)
+            self._lift_symbols_in_tensor(t._values(), src)
+        elif t.layout in {torch.sparse_csr, torch.sparse_bsr}:
+            self._lift_symbols_in_tensor(t.crow_indices(), src)
+            self._lift_symbols_in_tensor(t.col_indices(), src)
+        elif t.layout in {torch.sparse_csc, torch.sparse_bsc}:
+            self._lift_symbols_in_tensor(t.ccol_indices(), src)
+            self._lift_symbols_in_tensor(t.row_indices(), src)
+        if is_traceable_wrapper_subclass(t):
+            attrs, ctx = t.__tensor_flatten__()
+            for attr in attrs:
+                inner_t = getattr(t, attr)
+                self._lift_symbols_in_tensor(
+                    inner_t, AttrSource(src, attr) if src is not None else None
+                )
+
+    # Lookup the proxy in current tracer for each symbol in expressions of s,
+    # See Note [Auto lift basic free symbols when create_graph_input]
+    def lookup_bounded_symbols(
```

Comment:
"Returns a mapping from bound symbols to their proxies and a set of unbound symbols"